be/src/exec/pipeline/scan/connector_scan_operator.cpp

```diff
@@ -480,47 +477,25 @@ int ConnectorScanOperator::available_pickup_morsel_count() {
 
     // adjust io tasks according information collected
     P.cs_gen_chunks_time = (now - P.op_start_time - P.cs_total_halt_time);
+
     // chunk sources in each scan operator don't generate chunks in uniform speed
     // but chunks are distributed evenly to scan operator. So we I think we'd better not to
     // use chunks generated by chunks source issues by this scan operator, but to use
     // chunks generated by all scan operators. A approximate value is to use `op_pull_chunks`
     // produced chunks per 10ms, but cs running time is microsecond unit.
-    double balanced_cs_speed = _op_pull_chunks * 10000.0 / (P.cs_gen_chunks_time + 1);
+    double source_speed = _op_pull_chunks * 1e4 / (P.cs_gen_chunks_time + 1);
     // consumed chunks per 10ms, but op_running time is nanosecond unit.
-    double op_speed = _op_pull_chunks * 10000000.0 / (_op_running_time_ns + 1);
-    // `cs_speed` is speed in this single scan operator.
-    int64_t cs_pull_chunks = P.cs_pull_chunks.load();
-    double cs_speed = cs_pull_chunks * 10000.0 / (P.cs_gen_chunks_time + 1);
+    double operator_speed = _op_pull_chunks * 1e7 / (_op_running_time_ns + 1);
+
     // chunk source: total io time and running time.
     // we can see if this is slow device. io_latency in ms unit.
-    int64_t cs_total_io_time = P.cs_total_io_time.load();
-    int64_t cs_total_scan_bytes = P.cs_total_scan_bytes.load();
-    // how many 1MB read.
-    double norm_io_count = (cs_total_scan_bytes * 1.0) / (1024 * 1024);
-    double io_latency = cs_total_io_time * 0.000001 / norm_io_count;
+    double cs_total_io_time = P.cs_total_io_time.load() * 1e-6;
+    double cs_total_scan_bytes = P.cs_total_scan_bytes.load() / (1024 * 1024);
+    double io_latency = cs_total_io_time / (cs_total_scan_bytes + 1e-3);
```

Comment:
what's this 1e-3 for? avoid div-zero error?

Reply:
yes.